### PR TITLE
Source code and tests for Playlist and Song data structures

### PR DESCRIPTION
**Pull request by nathanduchesne**
If applied, this commit will add the possibility of creating playlists, which contain songs. Songs are equal if their artist name and title are identical, because we can safely assume an artist doesn't have two songs with the same name. Playlists can be added to one another.

### CODE REVIEW

**Comment by nathanduchesne** [comment]
Works now!

File: app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class Playlist (playListName: String, songSet: Set<Song>, genreType: Genre){
+    var name: String = playListName
+    private val songs: MutableSet<Song>
+    var genre: Genre = genreType
+
+    init {
+        val result = mutableSetOf<Song>()
+        for (song in songSet){
```
Comment (VictorSchneuwly) [inline app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt:8]: Why don't you use '.addAll(songSet)' here ?

File: app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    fun removeSongs(removedSongs: Set<Song>){
+        songs.removeAll(removedSongs)
+    }
+
+    fun combinePlaylists(other: Playlist, newName: String = name, newGenre: Genre = Genre.NONE) {
+        name = newName
+        genre = newGenre
+        for (song in other.songs) {
```
Comment (VictorSchneuwly) [inline app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt:36]: Same here, why not use 'addAll' ?

File: app/src/main/java/com/github/fribourgsdp/radio/Song.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import java.util.*
+
+class Song (songName: String, artistName: String, songLyrics: String) {
+
+    var name: String = reformat_song_name(songName)
+        get() {return field}
+        set(value) {field = reformat_song_name(value)}
+
+    var artist: String = artistName
+    var lyrics: String = songLyrics
```
Comment (VictorSchneuwly) [inline app/src/main/java/com/github/fribourgsdp/radio/Song.kt:10]: Why not make all of these as `val` to make a `Song` immutable?

Comment (nathanduchesne) [reply app/src/main/java/com/github/fribourgsdp/radio/Song.kt:10]: For custom playlists made by users I thought it would be nice to allow users the chance to change the song name.

File: app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt (hunk 2 above)
Comment (nathanduchesne) [reply app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt:36]: Same as above, I did this to avoid the fact that if the owner of the 'other' playlist changes a song name, it doesn't change in our new playlist that we combined previously.

File: app/src/main/java/com/github/fribourgsdp/radio/Song.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import java.util.*
+
+class Song (songName: String, artistName: String, songLyrics: String) {
+
+    var name: String = reformat_song_name(songName)
+        get() {return field}
+        set(value) {field = reformat_song_name(value)}
+
+    var artist: String = artistName
```
Comment (Flamantrose321) [inline app/src/main/java/com/github/fribourgsdp/radio/Song.kt:9]: Do you not reformat the artist name?

File: app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            songs.add(Song(song.name, song.artist, song.lyrics))
+        }
+    }
+
+    fun getSongs(): MutableSet<Song> {
+        return songs.toMutableSet()
```
Comment (shanzhaiii) [inline app/src/main/java/com/github/fribourgsdp/radio/Playlist.kt:44]: Would it not be better to only return an immutable copy of the playlist?

File: build.gradle [hunk 6]
```diff
@@ -5,7 +5,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:7.0.3"
```
Comment (shanzhaiii) [inline build.gradle:8]: Why was this changed?
